clang-tools-extra/clang-tidy/ClangTidy.cpp

```diff
@@ -449,6 +479,10 @@ ClangTidyASTConsumerFactory::createASTConsumer(
   }
 
   std::vector<std::unique_ptr<ASTConsumer>> Consumers;
+
+  if (!Context.getOptions().SystemHeaders.value_or(false))
+    Consumers.push_back(std::make_unique<IgnoreSystemHeadersConsumer>());
+
   if (!Checks.empty())
     Consumers.push_back(Finder->newASTConsumer());
```

Comment:
I prefer to do like this
```c++
if (!Checks.empty())
  if (!Context.getOptions().SystemHeaders.value_or(false))
    Consumers.push_back(std::make_unique<IgnoreSystemHeadersConsumer>(Finder->newASTConsumer()));
  else
    Consumers.push_back(Finder->newASTConsumer());
```
IgnoreSystemHeadersConsumer will skip all top level decl in system header

Reply:
Hmm, this code does not compile? `IgnoreSystemHeadersConsumer` does not have a constructor accepting a `unique_ptr<ASTConsumer>`?

There's also duplication in creating the `Finder->newASTConsumer()`, so I don't see how this solution would be preferable. I can certainly move the `IgnoreSystemHeadersConsumer` inside the `if(!Checks.empty())` though.